### PR TITLE
Fix relationals builtin translation bug

### PR DESCRIPTION
**Pull request by wwXing0**
According OpenCL spec, the relationals builtin function shall return a '1' if the specified relation is true for scalar argument types. The functions shall return a '-1' (i.e. all bits set) if the specified relation is true for vector argument types.

### CODE REVIEW

File: test/transcoding/relationals_select.ll [hunk 1]
```diff
@@ -0,0 +1,130 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -o %t.spv
+; RUN: spirv-val %t.spv
```
Comment (MrSidims) [inline test/transcoding/relationals_select.ll:3]: Suggest to remove spirv-val for now.
It errors with
error: line 74: All OpVariable instructions in a function must be the first instructions in the first block.
  %arg1 = OpVariable %_ptr_Function_v4float Function

and it relates to https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/1883

File: lib/SPIRV/OCLToSPIRV.cpp [hunk 2]
```diff
@@ -1137,11 +1137,14 @@ void OCLToSPIRVBase::visitCallRelational(CallInst *CI,
...
-    return Builder.CreateSelect(NewCI, Constant::getAllOnesValue(CI->getType()),
-                                Constant::getNullValue(CI->getType()));
-  });
+  mutateCallInst(CI, OC).changeReturnType(
+      BoolTy, [=](IRBuilder<> &Builder, CallInst *NewCI) {
+        Value *TruOp = CI->getType()->isVectorTy()
+                           ? Constant::getAllOnesValue(CI->getType())
+                           : getInt32(M, 1);
+        return Builder.CreateSelect(NewCI, TruOp,
+                                    Constant::getNullValue(CI->getType()));
```
Comment (MrSidims) [inline lib/SPIRV/OCLToSPIRV.cpp:1146]: guess, there is a typo
```suggestion
      BoolTy, [=](IRBuilder<> &Builder, CallInst *NewCI) {
        Value *TrueOp = CI->getType()->isVectorTy()
                           ? Constant::getAllOnesValue(CI->getType())
                           : getInt32(M, 1);
        return Builder.CreateSelect(NewCI, TrueOp,
                                    Constant::getNullValue(CI->getType()));
```

**Review by MrSidims — COMMENTED** [review]
LGTM, thanks for adding the test!

**Comment by wwXing0** [comment]
Hi~@MrSidims , this patch seems irrelative with the failure，do you have any idea about the failure?

**Comment by MrSidims** [comment]
@wwXing0 hi, there failures are relate to llvm.org change and are fixed by a patch from @svenvh

**Review by svenvh — APPROVED** [review]

**Review by MrSidims — APPROVED** [review]

**Pull request merged**